addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -76,7 +76,7 @@ var CourseJoinWidget = publicWidget.Widget.extend({
                 url += '?fullscreen=1';
             }
         } else {
-            url = `/slides/${this.channel.channelId}`;
+            url = `/slides/${encodeURIComponent(this.channel.channelId)}`;
         }
         document.location = _.str.sprintf('/web/login?redirect=%s', encodeURIComponent(url));
     },
```

Comment:
Indeed, and this is intentional: the `url` variable is used as a parameter inside another url, so it has to be escaped as well, therefore its `this.channel.channelId` will be escaped twice, so that the `/web/login/` page will extract a correctly escaped URL from the redirect parameter.

Here, if a silly `channelId` is `???`:
`url` is: `/slides/%3F%3F%3F`
`document.location` is `/web/login?redirect=%2Fslides%2F%253F%253F%253F`
Upon fulfilling the login, the `redirect` will be parsed as `decodeURIComponent("%2Fslides%2F%253F%253F%253F")` which is `/slides/%3F%3F%3F` onto which the browser will be redirected, and this is the right way to represent `???` inside a URL path (otherwise the first question mark would start the parameters part of the URL)

Reply:
After testing, youku want its untouched string - so I restored the initial code (it makes sense, the text put there is already coming from a truncated URL - so its content is already an URL content, it should not be escaped...)

Rhaaa... it sometimes gets confusing.